src/raise.cr

```diff
@@ -37,6 +37,58 @@ private struct LEBReader
   end
 end
 
+private def traverse_eh_table(leb, start, ip, actions)
+  # Ref: https://chromium.googlesource.com/native_client/pnacl-libcxxabi/+/master/src/cxa_personality.cpp
+
+  throw_offset = ip - 1 - start
+  # puts "Personality - actions : #{actions}, start: #{start}, ip: #{ip}, throw_offset: #{throw_offset}"
+
+  lp_start_encoding = leb.read_uint8 # @LPStart encoding
+  if lp_start_encoding != 0xff_u8
+    LibC.dprintf 2, "Unexpected encoding for LPStart: #{lp_start_encoding}\n"
+    LibC.exit 1
+  end
+
+  if leb.read_uint8 != 0xff_u8 # @TType encoding
+    leb.read_uleb128           # @TType base offset
+  end
+
+  cs_encoding = leb.read_uint8 # CS Encoding (1: uleb128, 3: uint32)
+  if cs_encoding != 1 && cs_encoding != 3
+    LibC.dprintf 2, "Unexpected CS encoding: #{cs_encoding}\n"
+    LibC.exit 1
+  end
+
+  cs_table_length = leb.read_uleb128 # CS table length
+  cs_table_end = leb.data + cs_table_length
+
+  while leb.data < cs_table_end
+    cs_offset = cs_encoding == 3 ? leb.read_uint32 : leb.read_uleb128
+    cs_length = cs_encoding == 3 ? leb.read_uint32 : leb.read_uleb128
+    cs_addr = cs_encoding == 3 ? leb.read_uint32 : leb.read_uleb128
+    action = leb.read_uleb128
+    # puts "cs_offset: #{cs_offset}, cs_length: #{cs_length}, cs_addr: #{cs_addr}, action: #{action}"
+
+    if cs_addr != 0
+      if cs_offset <= throw_offset && throw_offset <= cs_offset + cs_length
+        if actions.includes? LibUnwind::Action::SEARCH_PHASE
+          # puts "found"
+          return LibUnwind::ReasonCode::HANDLER_FOUND
+        end
+
+        if actions.includes? LibUnwind::Action::HANDLER_FRAME
+          unwind_ip = start + cs_addr
+          yield unwind_ip
+          # puts "install"
+          return LibUnwind::ReasonCode::INSTALL_CONTEXT
+        end
+      end
+    end
+  end
+
+  return nil
```

Comment:
Reduntant line.

Reply:
It's redundant for the compiler. It's not redundant for a human reading the code.